lib/index.js

```diff
@@ -5,7 +5,63 @@ const spawn = require('child_process').spawn;
 const spawnSync = require('child_process').spawnSync;
 const JSON = require('./json-buffer');
 const isElectron = require('is-electron');
+const fs = require('fs');
+const os = require('os');
+const index = [0];
+const timeStamp = new Date().toISOString();
+const net = require('net');
+let clientConnected = false;
+let client;
+const READFILETIMELIMIT = 5000;
+const debugMode = false;
 
+const responseFolder = path.join(os.tmpdir(), 'sync_rpc_responses');
+if (!fs.existsSync(responseFolder)) {
+  fs.mkdirSync(responseFolder, {recursive: true});
+}
+
+const responseFile = path.join(responseFolder, `responseFile.txt`);
+
+function timeLimitExceeded(startTime) {
+  return startTime + READFILETIMELIMIT < Date.now();
+}
+
+function responseFileReadOnly() {
+  const mode = fs.statSync(responseFile).mode;
+  return (mode & parseInt('777', 8)).toString(8) == '444';
+}
+
+function responseFileReadyForReading() {
+  return fs.existsSync(responseFile) && responseFileReadOnly();
+}
+
+function readResponseFile() {
+  const startTime = Date.now();
+  while (!responseFileReadyForReading()) {
+    // if (timeLimitExceeded(startTime)) {
+    //   return '';
+    // }
+  }
+  const res = fs.readFileSync(responseFile);
+  fs.unlinkSync(responseFile);
+  return res;
+}
+
+const outFolder = path.join(os.tmpdir(), 'sync_rpc_log');
+if (debugMode && !fs.existsSync(outFolder)) {
+  fs.mkdirSync(outFolder, {recursive: true});
+}
+
+const localClientLog = path.join(outFolder, `client_logs_${timeStamp}.txt`);
```

Comment:
what is the purpose of this. Should this be in debugMode only?

Reply:
Yes, only in debugMode it is being used. Otherwise, it is just created but never used.